source/simulator/helper_functions.cc

```diff
@@ -1694,72 +1686,82 @@ namespace aspect
     // So even though we touch some DoF more than once, we always start from the same value, compute the
     // same value, and then overwrite the same value in distributed_vector.
     // TODO: make this more efficient.
+
+    using VectorType = Vector<double>;
+    SUNDIALS::ARKode<VectorType>::AdditionalData data (0.0, time_step, 0.001*time_step, 0.1*time_step, 1.e-6*time_step);
+    SUNDIALS::ARKode<VectorType> ode(data);
+
     for (const auto &cell : dof_handler.active_cell_iterators())
       if (cell->is_locally_owned())
         {
           fe_values_C.reinit (cell);
           in_C.reinit(fe_values_C, cell, introspection, solution);
 
-          if (temperature_and_composition_use_same_fe == false)
-            {
-              fe_values_T.reinit (cell);
-              in_T.reinit(fe_values_T, cell, introspection, solution);
-            }
-
           std::vector<std::vector<double>> accumulated_reactions_C (quadrature_C.size(),std::vector<double> (introspection.n_compositional_fields));
           std::vector<double> accumulated_reactions_T (quadrature_T.size());
 
-          // Make the reaction time steps: We have to update the values of compositional fields and the temperature.
-          // Because temperature and composition might use different finite elements, we loop through their elements
-          // separately, and update the temperature and the compositions for both.
-          // We can reuse the same material model inputs and outputs structure for each reaction time step.
-          // We store the computed updates to temperature and composition in a separate (accumulated_reactions) vector,
-          // so that we can later copy it over to the solution vector.
-          for (unsigned int i=0; i<number_of_reaction_steps; ++i)
-            {
-              // Loop over composition element
-              material_model->fill_additional_material_model_inputs(in_C, solution, fe_values_C, introspection);
-
-              material_model->evaluate(in_C, out_C);
-              heating_model_manager.evaluate(in_C, out_C, heating_model_outputs_C);
-
-              for (unsigned int j=0; j<dof_handler.get_fe().base_element(introspection.base_elements.compositional_fields).dofs_per_cell; ++j)
+          std::vector<std::vector<double>> initial_values_C (quadrature_C.size(),std::vector<double> (introspection.n_compositional_fields));
```

Comment:
Move the declaration of this variable out of the loop over all cells. This way it needs to allocate memory only once.